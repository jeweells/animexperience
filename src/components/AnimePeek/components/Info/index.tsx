import React from 'react'
import styled from 'styled-components'
import { FColG16, FRowG8 } from '../../../../atoms/Layout'

export type InfoProps = {
    icon?: React.ReactNode
    title: string
}
const Title = styled(FRowG8)`
    gap: 8px;
    align-items: center;
    font-weight: 500;
    font-size: 1.1rem;
`
const Value = styled.div`
    color: rgba(233, 235, 240, 0.8);
    font-size: 0.9rem;
`
export const Info: React.FC<InfoProps> = React.memo(
    ({ title, icon = null, children }) => {
        return (
            <FColG16>
                <Title as={'h5'}>
                    {icon}
                    {title}
                </Title>
                <Value>{children}</Value>
            </FColG16>
        )
    },
)

Info.displayName = 'Info'

export default Info
