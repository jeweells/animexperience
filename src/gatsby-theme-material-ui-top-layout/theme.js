import { createTheme } from '@material-ui/core/styles'

export const mainColor = 'rgb(105, 190, 213)'

const theme = createTheme({
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    overrides: {
        MuiButtonBase: {
            root: {
                borderRadius: undefined,
            },
        },
        MuiButton: {
            root: {
                color: '#e9ebf0',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                height: 36,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },

            text: {
                color: '#e9ebf0',
                textTransform: 'none',
                padding: '8px 12px',
            },
        },
        MuiIconButton: {
            root: {
                flexShrink: 0,
                height: 36,
                minWidth: 36,
                color: '#e9ebf0',
                padding: '8px 12px',
                backgroundColor: '#292d33',
                marginBottom: 0,
                fontWeight: 'normal',
                textAlign: 'center',
                verticalAlign: 'middle',
                outline: '0 !important',
                whiteSpace: 'nowrap',
                border: 'none',
                userSelect: 'none',
                fontSize: 14,
                lineHeight: '1.42857143',
                borderRadius: 6,
                '&:hover': {
                    backgroundColor: '#3c3f43',
                },
            },
        },
        MuiCssBaseline: {
            '@global': {
                html: {
                    WebkitFontSmoothing: 'auto',
                },
                body: {
                    color: undefined,
                    margin: undefined,
                    fontSize: undefined,
                    fontFamily: undefined,
                    fontWeight: undefined,
                    backgroundColor: undefined,
                    lineHeight: undefined,
                },
            },
        },
        MuiSkeleton: {
            root: {
                backgroundColor: 'rgba(255, 255, 255, 0.11)',
            },
            pulse: {
                animation: 'MuiSkeleton-keyframes-pulse 3s ease-in 0.5s infinite',
            },
        },
    },
})

export default theme
