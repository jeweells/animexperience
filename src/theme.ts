import { createTheme } from '@mui/material/styles'

const theme = createTheme({
    breakpoints: {
        values: {
            xs: 0,
            sm: 500,
            md: 800,
            lg: 1000,
            xl: 1200,
        },
    },
    typography: {
        fontFamily: ['Quicksand', 'sans-serif'].join(','),
    },
    palette: {
        mode: 'dark',
        primary: {
            main: '#292d33',
            contrastText: '#e9ebf0',
        },
        secondary: {
            main: '#3c3f43',
        },
        text: {
            primary: '#e9ebf0',
            secondary: '#e9ebf0',
            disabled: '#ffffff82',
        },
    },
    components: {
        MuiSvgIcon: {
            styleOverrides: {
                root: {
                    fontSize: 'inherit',
                },
            },
        },
        MuiButtonBase: {
            styleOverrides: {
                root: {
                    borderRadius: undefined,
                    textTransform: 'unset',
                },
            },
        },
        MuiButton: {
            styleOverrides: {
                root: {
                    color: '#e9ebf0',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: 14,
                    lineHeight: '1.42857143',
                    textTransform: 'unset',
                    borderRadius: 6,
                    height: 36,
                    '&:hover': {
                        outline: 'none',
                        border: 'none',
                        backgroundColor: '#3c3f43',
                    },
                },
                text: {
                    color: '#e9ebf0',
                    textTransform: 'none',
                    padding: '8px 12px',
                },
            },
        },
        MuiIconButton: {
            styleOverrides: {
                root: {
                    flexShrink: 0,
                    height: 36,
                    minWidth: 36,
                    color: '#e9ebf0',
                    padding: '8px 6px',
                    backgroundColor: '#292d33',
                    marginBottom: 0,
                    fontWeight: 'normal',
                    textAlign: 'center',
                    verticalAlign: 'middle',
                    outline: '0 !important',
                    whiteSpace: 'nowrap',
                    border: 'none',
                    userSelect: 'none',
                    fontSize: '1.2rem',
                    lineHeight: '1.42857143',
                    borderRadius: 6,
                    '&:hover': {
                        backgroundColor: '#3c3f43',
                    },
                    '&.Mui-disabled': {
                        color: 'rgb(255 255 255 / 25%)',
                    },
                },
            },
        },
        MuiSkeleton: {
            defaultProps: {
                animation: 'pulse',
            },
            styleOverrides: {
                root: {
                    backgroundColor: 'rgba(255, 255, 255, 0.11)',
                },
                pulse: {
                    animationDuration: '1.5s 0.5s',
                },
            },
        },
        MuiCssBaseline: {
            styleOverrides: `
                * {
                    backface-visibility: hidden;
                    margin: 0;
                    padding: 0;
                    box-sizing: border-box;
                }
                html {
                    -webkit-font-smoothing: auto;
                }
                h6 {
                    margin: 0;
                }
                body {
                    background-color: #0F131A;
                }
                ::-webkit-scrollbar {
                    width: 10px;
                }
                ::-webkit-scrollbar-track {
                    background: transparent;
                }
                ::-webkit-scrollbar-thumb {
                    background: #25282e;
                },
                ::-webkit-scrollbar-thumb:hover {
                    background: #393a44;
                }
            `,
        },
    },
})

export default theme
