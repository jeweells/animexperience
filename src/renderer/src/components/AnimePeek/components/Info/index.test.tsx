import { ThemeProvider } from '@mui/material/styles'
import Info, { InfoProps } from './index'
import theme from '../../../../theme'

describe('Info', () => {
  let props: InfoProps

  const getComponent = (override: Partial<InfoProps> = {}) => {
    return (
      <ThemeProvider theme={theme}>
        <Info {...props} {...override} />
      </ThemeProvider>
    )
  }

  beforeEach(() => {
    props = {
      title: 'Sample title',
      icon: <div>Sample icon inside div</div>,
      children: <div>Sample children inside div</div>
    }
  })

  it('renders default', async () => {
    const wrapper = render(getComponent())
    expect(wrapper.asFragment()).toMatchSnapshot()
  })
})
