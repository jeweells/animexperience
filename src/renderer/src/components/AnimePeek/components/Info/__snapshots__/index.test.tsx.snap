// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info renders default 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 16px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  gap: 8px;
  gap: 8px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 500;
  font-size: 1.1rem;
}

.emotion-2 {
  color: rgba(233, 235, 240, 0.8);
  font-size: 0.9rem;
}

<div
    class="emotion-0"
  >
    <h5
      class="emotion-1"
    >
      <div>
        Sample icon inside div
      </div>
      Sample title
    </h5>
    <div
      class="emotion-2"
    >
      <div>
        Sample children inside div
      </div>
    </div>
  </div>
</DocumentFragment>
`;
